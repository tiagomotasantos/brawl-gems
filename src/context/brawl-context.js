import React from 'react'

const BrawlContext = React.createContext()

export { BrawlContext as default }