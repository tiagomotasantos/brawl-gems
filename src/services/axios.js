import axios from 'axios';

const axiosInstance = axios.create({
    baseURL: 'https://api.brawlstats.com/'
});

export const login = () => {
    axiosInstance.get('v6/auth/exchange-key').then(({ data }) => {
        axiosInstance.defaults.headers.common['Authorization'] = `Bearer ${data}`;;
    });
}

export default axiosInstance;