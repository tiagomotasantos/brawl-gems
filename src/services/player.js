
import axios from './axios';

export const getPlayer = (tag) => {
    return axios.get(`player?tag=${tag}`);
}