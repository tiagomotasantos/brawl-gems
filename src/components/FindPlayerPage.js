import React, { useState } from 'react';
import { getPlayer } from '../services/player';

const FindPlayerPage = () => {
    const [tag, setTag] = useState('');
    const onSubmit = (e) => {
        e.preventDefault();

        getPlayer(tag).then(res => console.log(res));

        //setTag('');
    }

    return (    
        <div>
            <div>Find player by tag</div>
            <form onSubmit={onSubmit}>
                <input 
                    type="text"
                    value={tag}
                    placeholder="Player tag"
                    autoFocus
                    onChange={(e) => setTag(e.target.value)}
                />
                <button>Find</button>
            </form>
        </div>
    );
};

export default FindPlayerPage;