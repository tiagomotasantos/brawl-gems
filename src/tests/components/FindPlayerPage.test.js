import React from 'react';
import { shallow } from 'enzyme';
import FindPlayerPage from '../../components/FindPlayerPage';
import player from '../fixtures/player';
import * as BrawlContext from '../../context/brawl-context';
import * as Services from '../../services/player';

let history, wrapper, dispatch;

beforeEach(() => {
    dispatch = jest.fn();
    history = { push: jest.fn() };
    BrawlContext.default = React.createContext({ dispatch });
    wrapper = shallow(<FindPlayerPage history={history}/>);
});

test('should render FindPlayerPage', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should call get player', (done) => {
    const preventDefault = jest.fn();
    const getPlayerPromise = new Promise((resolve) => (resolve(player)));
    Services.getPlayer = jest.fn().mockReturnValue(getPlayerPromise);

    wrapper.find('input').prop('onChange')({ target: { value: player.tag }});
    wrapper.find('form').prop('onSubmit')({ preventDefault });
    
    expect(Services.getPlayer).toHaveBeenLastCalledWith(player.tag);
    
    getPlayerPromise.then(() => {
        expect(dispatch).toHaveBeenLastCalledWith( { type: 'SET_PLAYER', player });
        expect(history.push).toHaveBeenLastCalledWith(`/players/${player.tag}`);
        done();
    });
});