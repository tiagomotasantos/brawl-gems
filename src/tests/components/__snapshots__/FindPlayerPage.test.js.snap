// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FindPlayerPage 1`] = `
<div
  className="content-container find-player__content"
>
  <img
    alt="Brawl Stars"
    className="find-player__logo"
    src="/images/full_logo.png"
  />
  <h3
    className="find-player__title"
  >
    Player Analysis
  </h3>
  <form
    className="find-player__form"
    onSubmit={[Function]}
  >
    <div
      className="find-player__hashtag"
    >
      #
    </div>
    <input
      autoFocus={true}
      className="input"
      onChange={[Function]}
      placeholder="Player tag"
      type="text"
      value=""
    />
    <button
      className="find-player__button"
    >
      <img
        alt="Search"
        src="/images/search.png"
      />
    </button>
  </form>
</div>
`;
